consensus/spos/bls/v2/subroundEndRound.go

```diff
@@ -740,38 +521,34 @@ func (sr *subroundEndRound) computeAggSigOnValidNodes() ([]byte, []byte, error)
 	return bitmap, sig, nil
 }
 
-func (sr *subroundEndRound) createAndBroadcastHeaderFinalInfoForKey(signature []byte, bitmap []byte, leaderSignature []byte, pubKey []byte) bool {
-	index, err := sr.ConsensusGroupIndex(string(pubKey))
-	if err != nil {
-		log.Debug("createAndBroadcastHeaderFinalInfoForKey.ConsensusGroupIndex", "error", err.Error())
-		return false
-	}
-
+func (sr *subroundEndRound) createAndBroadcastProof(signature []byte, bitmap []byte) *block.HeaderProof {
 	headerProof := &block.HeaderProof{
-		AggregatedSignature: signature,
 		PubKeysBitmap:       bitmap,
+		AggregatedSignature: signature,
 		HeaderHash:          sr.GetData(),
 		HeaderEpoch:         sr.GetHeader().GetEpoch(),
+		HeaderNonce:         sr.GetHeader().GetNonce(),
+		HeaderShardId:       sr.GetHeader().GetShardID(),
 	}
 
-	sr.BroadcastMessenger().PrepareBroadcastEquivalentProof(headerProof, index, pubKey)
-	log.Debug("step 3: block header proof has been sent to delayed broadcaster",
+	err := sr.BroadcastMessenger().BroadcastEquivalentProof(headerProof, []byte(sr.SelfPubKey()))
+	if err != nil {
+		return nil
+	}
+
+	log.Debug("step 3: block header proof has been sent",
 		"PubKeysBitmap", bitmap,
-		"AggregateSignature", signature,
-		"LeaderSignature", leaderSignature,
-		"Index", index)
+		"AggregateSignature", signature)
 
-	return true
+	return headerProof
 }
 
 func (sr *subroundEndRound) createAndBroadcastInvalidSigners(invalidSigners []byte) {
-	// TODO[cleanup cns finality]: remove the leader check
-	isEquivalentMessagesFlagEnabled := sr.EnableEpochsHandler().IsFlagEnabledInEpoch(common.EquivalentMessagesFlag, sr.GetHeader().GetEpoch())
-	if !sr.IsSelfLeader() && !isEquivalentMessagesFlagEnabled {
+	if !sr.IsSelfLeader() {
```

Comment:
I don't thinks this would be ok, as the invalid signers should be broadcast by the one who also sent the proof, not by the leader